x/stakeibc/keeper/community_pool_test.go

```diff
@@ -0,0 +1,146 @@
+package keeper_test
+
+import (
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/gogoproto/proto"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+
+	epochtypes "github.com/Stride-Labs/stride/v14/x/epochs/types"
+	icqtypes "github.com/Stride-Labs/stride/v14/x/interchainquery/types"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/keeper"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/types"
+)
+
+func (s *KeeperTestSuite) SetupQueryCommunityPoolBalance(icaAccountType types.ICAAccountType) types.HostZone {
+	// We need to register the transfer channel to initialize the light client state
+	s.CreateTransferChannel(HostChainId)
+
+	// Craete host zone
+	// We must use valid addresses for each ICA since they're serialized for the query request
+	depositAddress := s.TestAccs[0]
+	returnAddress := s.TestAccs[1]
+	hostZone := types.HostZone{
+		ChainId:                        HostChainId,
+		ConnectionId:                   ibctesting.FirstConnectionID,
+		CommunityPoolDepositIcaAddress: depositAddress.String(),
+		CommunityPoolReturnIcaAddress:  returnAddress.String(),
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	// Create epoch tracker for timeout
+	epochEndTime := uint64(1000)
```

Comment:
suggestion: instead of using an absolute value here, we should use the ctx.BlockTime and add something to it.

This is because in the query function we do 
`timeout := time.Unix(0, int64(strideEpochTracker.NextEpochStartTime))
timeoutDuration := timeout.Sub(ctx.BlockTime())`

So if this test is working, it is because BlockTime() is 0 in the test framework, just about any other BlockTime would lead to a negative duration

Reply:
So this was actually intentional, because there isn't a validation that the time is greater than the current block time like there is in transfer (so I was trying to indicate in the test that a dummy value would work).

That said, I realize I should be validating the timeout was set in the query object and it doesn't make sense to use a negative time. I'll give it another pass!

Reply:
just pushed up a new attempt - lmk what you think